herbert_core/utilities/classes/@IX_data_2d/noisify.m

```diff
@@ -16,5 +16,7 @@
 %
 wout=w;
 for i=1:numel(w)
-    [wout(i).signal,wout(i).error]=noisify(w(i).signal,w(i).error,varargin{:});
+    [wout(i).signal, errsqr] = noisify (...
```

Comment:
Rather than maintaining N copies of this, add this to the IX_dataset base-class?

Reply:
In a parallel branch I am doing a complete overhaul of IX_dataset and all its children, with one of the goals to replace the 1D,2D,3D versions of a method with just one generic n-dimensional method; there are many such methods, not just noisify. here my intention is just to get the noisify methods to work correctly as swiftly as possible.